packages/read/src/common.ts

```diff
@@ -0,0 +1,61 @@
+import { GraphData } from '@mindgraph/types';
+
+export const MARKDOWN_EXTENSION = '.md';
+export const HIDDEN_FILES_REGEX = /^\./;
+export const LINK_CONTENT_REGEX = /\]\((.*?)\)/g;
+
+export function formatGraphForTestSnapshot(data: GraphData) {
+  return {
+    nodes: data.nodes
+      .map((n) => ({ name: n.name, linkCount: n.linkCount }))
+      .sort((a, b) => (a.name > b.name ? 1 : -1)),
+    links: data.links
+      .map((l) => ({
+        source: l.source.split('/').at(-1)!,
+        target: l.target.split('/').at(-1),
+      }))
+      .sort((a, b) => (a.source > b.source ? 1 : -1)),
+  };
+}
+
+export function extractLinksFromLine(
+  line: string,
+  filePath: string,
+): GraphData['links'] {
+  const formattedLinks = [];
+  const links = line.match(LINK_CONTENT_REGEX) || [];
+
+  for (const link of links) {
+    const path = LINK_CONTENT_REGEX.exec(link)?.at(1);
```

Comment:
I think there are some unintended consequences of using this `exec` along with the global flag on this particular Regex, see: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec#description

I wrote a fix (I think), here: https://github.com/jollyjerr/mindgraph/pull/29/files#diff-09c58197e1dfd80696e23cc9172cbf7fe3cc9f9c048c349a133f49256705e4edR26, and also added some unit tests for this function

Reply:
Woah I had no idea this statefulness was a thing 😮, I'll merge your new implementation for sure! Thank you for doing that!